assets/src/application/services/toaster/utilities/getAnimation.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * adapted from:
+ * @link https://github.com/bmcmahen/toasted-notes/blob/master/src/Alert/Message.ts
+ */
+const getAnimation = ({ container, onRest, position }): any => {
+	const isFromTop = position === 'top-left' || position === 'top-right' || position === 'top';
+	return {
+		config: { mass: 1, tension: 185, friction: 26 },
+		from: {
+			opacity: 1,
+			height: 0,
+			transform: `translateY(${isFromTop ? '-100%' : 0}) scale(1)`,
+		},
+		enter: (): Function => (next: any): void =>
+			next({
+				opacity: 1,
+				height: container?.current?.getBoundingClientRect()?.height,
+				transform: `translateY(0) scale(1)`,
+			}),
+		leave: {
+			opacity: 0,
+			height: 0,
+			transform: `translateY(0 scale(0.9)`,
+		},
+		onRest,
```

Comment:
very interesting approach

Reply:
I am 1000% sure that with raw css this would have been much messier and less maintainable

Reply:
not my code btw so not going to take any credit for it... all i did was extract it into its own file for SRP purposes